tests/ord-aggregator/tests/handler_test.go

```diff
@@ -72,205 +77,251 @@ const (
+	for i := range buffer {
+		buffer[i] = letters[rand.Intn(len(letters))]
+	}
+	return string(buffer)
+}
 
-	bundlesAPIsNumberMap := make(map[string]int)
-	bundlesAPIsNumberMap[expectedBundleTitle] = expectedNumberOfAPIsInFirstBundle
-	bundlesAPIsNumberMap[secondExpectedBundleTitle] = expectedNumberOfAPIsInSecondBundle
+func TestORDAggregator(t *testing.T) {
+	ordConfigSecurity := &ordConfigSecurity{
+		Enabled:  false,
+		Username: randString(8),
+		Password: randString(8),
+	}
 
-	bundlesAPIsData := make(map[string][]string)
-	bundlesAPIsData[expectedBundleTitle] = []string{firstAPIExpectedTitle}
-	bundlesAPIsData[secondExpectedBundleTitle] = []string{firstAPIExpectedTitle}
+	defer func() { // Cleanup just in case (remove security)
+		ordConfigSecurity.Enabled = false
+		toggleORDConfigSecurity(t, testConfig.ExternalServicesMockBaseURL, ordConfigSecurity)
+	}()
+
+	for _, secured := range []bool{false, true} {
+		ordConfigSecurity.Enabled = secured
+		toggleORDConfigSecurity(t, testConfig.ExternalServicesMockBaseURL, ordConfigSecurity)
+
+		var appInput, secondAppInput, thirdAppInput directorSchema.ApplicationRegisterInput
+		t.Run(fmt.Sprintf("Verifying ORD Document to be valid [secured = %t]", secured), func(t *testing.T) {
+			if secured {
+				appInput = fixtures.FixSampleApplicationRegisterInputWithORDSecuredWebhooks(expectedSystemInstanceName, expectedSystemInstanceDescription, testConfig.ExternalServicesMockBaseURL, ordConfigSecurity.Username, ordConfigSecurity.Password)
+				appInput.Labels = directorSchema.Labels{applicationTypeLabelKey: []interface{}{testConfig.SecuredApplicationTypes[0]}}
+				secondAppInput = fixtures.FixSampleApplicationRegisterInputWithORDSecuredWebhooks(expectedSecondSystemInstanceName, expectedSecondSystemInstanceDescription, testConfig.ExternalServicesMockBaseURL, ordConfigSecurity.Username, ordConfigSecurity.Password)
+				secondAppInput.Labels = directorSchema.Labels{applicationTypeLabelKey: []interface{}{testConfig.SecuredApplicationTypes[0]}}
+				thirdAppInput = fixtures.FixSampleApplicationRegisterInputWithORDSecuredWebhooks(expectedThirdSystemInstanceName, expectedThirdSystemInstanceDescription, testConfig.ExternalServicesMockAbsoluteURL, ordConfigSecurity.Username, ordConfigSecurity.Password)
+				thirdAppInput.Labels = directorSchema.Labels{applicationTypeLabelKey: []interface{}{testConfig.SecuredApplicationTypes[0]}}
+			} else {
+				appInput = fixtures.FixSampleApplicationRegisterInputWithORDWebhooks(expectedSystemInstanceName, expectedSystemInstanceDescription, testConfig.ExternalServicesMockBaseURL)
+				secondAppInput = fixtures.FixSampleApplicationRegisterInputWithORDWebhooks(expectedSecondSystemInstanceName, expectedSecondSystemInstanceDescription, testConfig.ExternalServicesMockBaseURL)
+				thirdAppInput = fixtures.FixSampleApplicationRegisterInputWithORDWebhooks(expectedThirdSystemInstanceName, expectedThirdSystemInstanceDescription, testConfig.ExternalServicesMockAbsoluteURL)
+			}
```

Comment:
Can we encapsulate this `secured: true/false` in `fixtures.FixSampleApplicationRegisterInputWithORDWebhooks`
And don't have this branching here. The fixture can accept a bool parameter and create the relevant Webhook and labels based on the sceurity.